industrial_ci/src/tests/source_tests.sh

```diff
@@ -97,6 +97,96 @@ function run_clang_tidy_check {
     fi
 }
 
+function install_pylint {
+    local cmd=$1
+
+    ici_time_start "install_$cmd"
+    ici_quiet ici_install_pkgs_for_command "$cmd" "$cmd"
+    ici_time_end # install_$cmd
+}
+
+function run_pylint_file {
+    local _run_pylint_file_exit_code=$1; shift
+    local target_ws=$1; shift
+    local cmd=$1; shift
+    local file=$1; shift
+    local name=$1; shift
+    local pylint_args=("$@")
+    local status=0
+
+    ici_time_start "pylint_file_check_$name"
+
+    if ici_exec_in_workspace "$target_ws/install" "$target_ws" "$cmd" "${pylint_args[@]}" "$file"; then
+        ici_time_end
+    else
+        status=$?
+        # shellcheck disable=SC2140
+        eval "$_run_pylint_file_exit_code"="'$status'"
+        ici_time_end "${ANSI_YELLOW}" "$status"
+    fi
+}
+
+function run_pylint {
+    local -n _run_pylint_errors=$1; shift
+    local target_ws=$1; shift
+    local cmd=$1; shift
+    local pylint_args=("$@")
+
+    local -a pylint_find_pattern
+    # shellcheck disable=SC2016,SC2034
+    for p in $PYLINT_EXCLUDE; do pylint_find_pattern+=(-not -path '*$p*'); done
+    pylint_find_pattern+=(-type f -iname '*.py')
+
+    local path; path="$target_ws/src"
+
+    while read -r file; do
+        local name=${file#$path/}
+        local exit_code=0
+        run_pylint_file exit_code "$target_ws" "$cmd" "$file" "$name" "${pylint_args[@]}"
```

Comment:
Why did you factor this out into a function?
The call looks rather complicated, especially with `_run_pylint_errors`

Reply:
`run_pylint` runs pylint/pylint3 in general while `run_pylint_file` runs pylint/pylint3 on a single file....and then adds the file to the errors in case pylint reports an error....